### ISSUE TITLE
Support yearly/monthly archive folders

### ISSUE THREAD

**Issue by Eilyra** [k9mail/k-9#900]
Currently using Thunderbird as a desktop MUA, which supports automatically archiving messages into yearly or monthly subdirectories of the primary archive directory. Examples from Thunderbird below.

Yearly archive:
<img width="352" alt="archive" src="https://cloud.githubusercontent.com/assets/5540445/11223256/b7db0cf6-8d77-11e5-863e-3ae888beb6fc.png">

Monthly archive:
<img width="347" alt="archive_monthly" src="https://cloud.githubusercontent.com/assets/5540445/11223339/24320d14-8d78-11e5-9c5f-0a8fa4f6f4f4.png">

**Comment by guettli** [comment]
Yes, would be great to have this feature.
What can I do make this happen?

**Comment by jondo** [comment]
I would also need  yearly archive subfolders.

@guettli Your requested archive button is already available - I am using it all the time. Please ask in the [support mailing list](https://groups.google.com/forum/#!forum/k-9-mail) for how to configure it.

**Comment by guettli** [comment]
@jondo are the no docs which explain how to configure it?

**Comment by guettli** [comment]
... no progress here ... sad. Is there something I can do?

**Comment by philipwhiuk** [comment]
You could implement it @guettli. Short of that, it's fairly well described.

Complex features are generally developed by a developer who needs the feature. Personally speaking features I want / need is already a long enough list to keep me busy.

**Comment by Eilyra** [comment]
> As an aside, an archival service would need to be done quite efficiently - we don't have lots of spare CPU time to do this kind of thing.

Ah, there seems to be a miscommunication on my part on what is being requested. The feature is a tad simpler than that, it's merely extending the existing Archive command to file the message being archived into an appropriate subfolder of the archive folder, creating it as necessary. It wouldn't touch existant archived files, merely refile messages more intelligently. Consequently, CPU and bandwidth considerations should be minimal merely requiring some additional logic when the user presses the archive button.

A proper automated archival service does seem quite out of scope for a mobile email client.

**Comment by philipwhiuk** [comment]
Oh fair enough. Anyway, the point about developer priority still stands.

It's worth pointing out K-9 currently can't create folders, which would seem to be a good first step.

**Comment by guettli** [comment]
Yes creating folders is needed for this feature.

Why can't K9 create IMAP folders?

Is this difficult?

**Comment by philipwhiuk** [comment]
I don't think so - it's just never been a demand I guess. It looks fairly straightforward: https://tools.ietf.org/html/rfc3501#page-34

**Comment by Dinduks** [comment]
Given the huge leg work necessary to support folders creating, would it be possible to add an experimental version that only supports archiving into an existing folder? 

If the folder doesn't exist the user will have to create it manually through Thunderbird or some other interface. 

This manual folder creating must be done once a year and is much less taxing/annoying than having to do `Refile > Move > Select folder` for every single email.

For me personally this would make managing emails a breeze because for now it's such a pain to move each email manually.  
If this doesn't work my last resort will be switching to a flat archives structure.

What do you guys think?

**Comment by Miademora** [comment]
Something like a "quick move"/archive button would be amazing :)

**Comment by cketti** [comment]
K-9 Mail already offers a quick way to archive messages. The "Archive" action is located in the "Refile" menu when viewing a message. It can be promoted to a first level action under 'General settings > Display > Visible message actions'. See also: https://k9mail.app/documentation/settings/global#visible-message-actions

We use this issue tracker to track bugs and feature requests, not for support issues. In the future please use the [mailing list](http://groups.google.com/group/k-9-mail) for that.

**Comment by Dinduks** [comment]
@cketti the initial request in this issue was adding the support of archiving by year, which k9-mail doesn't support yet.
This sounds not realizable at the moment so I offered a more practical solution, which was marked "off-topic".  
Was it a mistake? If not, what's your take on it?

**Comment by cketti** [comment]
@Dinduks: Maybe I misunderstood how this feature is supposed to work. Do messages archived today end up in in a `2020-03` folder or does the message's date determine the folder the message is archived into?

In the first case, you can use the regular archive action and change the archive folder used by K-9 Mail each month. In the second case you'll have to wait until such a feature is implemented in K-9 Mail. "Archive into folder when that folder exists" is not a feature, it's a source of frustration and endless support requests. So no, this is not a practical solution.

**Comment by hockeymikey** [comment]
@cketti Emails would be archived based on the message date just like in Thunderbird (which I suppose one could have an option for based on the current time too incase someone wanted that too). Having archiving functionality without folder creation is still a feature :) just because you say its not doesn't make it so. Basically do you alot of work that will be needed for the feature minus the folder creation. That can be added in later when its there eadily into the existing code. You can put this feature under an experimental section too so users know its not fleshed out exactly yet. Can't believe this has been on the chopping block for half a decade, its such a key feature. Additional features would be ability for custom year/month or heck even day too formats. Allow the user to define a custom string if they wish since Thunderbird for example allows for this through a hidden pref (which an addon can easily change for the end user). Also I don't know the effects on threads off the top of my head. I know the Thunderbird archive function takes this into account from my readings.

**Comment by txtsd** [comment]
I've recently switched to K-9 Mail as my main mail client on Android.

I use the monthly folders archiving feature on Thunderbird on my desktop, and I need this feature to exist in K-9 Mail too.

Without this feature, I have to either wait until I get to my desktop to archive my mails, or if I do archive them from K-9, I have to re-archive them from the Archive(s) folder so they go to the correct monthly folder.

**Comment by Croydon** [comment]
Since we are getting closer to a Thunderbird for Android release, this might be a good time to tackle this feature in order to close the gap between desktop and mobile a bit further 🙂

**Comment by fnune** [comment]
Tried to implement this here https://github.com/thunderbird/thunderbird-android/pull/10113
